Jenkinsfile

```diff
@@ -0,0 +1,183 @@
+  ])
+])
+
+node('ubuntu-zion') {
+  def commitId, commitDate, version, imageId, branch, dockerImages
+  def organization = 'sonatype',
+      gitHubRepository = 'docker-nexus',
+      credentialsId = 'integrations-github-api',
+      imageName = 'sonatype/nexus',
+      archiveName = 'docker-nexus',
+      dockerHubRepository = 'nexus'
+  GitHub gitHub
+
+  try {
+    stage('Preparation') {
+      deleteDir()
+      OsTools.runSafe(this, "docker system prune -a -f")
+
+      def checkoutDetails = checkout scm
+
+      dockerImages = [
+        [ dockerFilePath: "${pwd()}/oss/Dockerfile", imageTag: "${imageName}:oss", flavor: "oss" ],
+        [ dockerFilePath: "${pwd()}/pro/Dockerfile", imageTag: "${imageName}:pro", flavor: "pro" ]
+      ]
+
+      branch = checkoutDetails.GIT_BRANCH == 'origin/master' ? 'master' : checkoutDetails.GIT_BRANCH
+      commitId = checkoutDetails.GIT_COMMIT
+      commitDate = OsTools.runSafe(this, "git show -s --format=%cd --date=format:%Y%m%d-%H%M%S ${commitId}")
+
+      OsTools.runSafe(this, 'git config --global user.email sonatype-ci@sonatype.com')
+      OsTools.runSafe(this, 'git config --global user.name Sonatype CI')
+
+      version = readVersion()
+
+      def apiToken
+      withCredentials([[$class: 'UsernamePasswordMultiBinding', credentialsId: credentialsId,
+                        usernameVariable: 'GITHUB_API_USERNAME', passwordVariable: 'GITHUB_API_PASSWORD']]) {
+        apiToken = env.GITHUB_API_PASSWORD
+      }
+      gitHub = new GitHub(this, "${organization}/${gitHubRepository}", apiToken)
+
+      if (params.nexus_repository_manager_version) {
+        stage('Update Repository Manager Version') {
+          OsTools.runSafe(this, "git checkout ${branch}")
+          dockerImages.each { updateRepositoryManagerVersion(it.dockerFilePath) }
+          version = params.nexus_repository_manager_version
+        }
+      }
+    }
+    stage('Build') {
+      gitHub.statusUpdate commitId, 'pending', 'build', 'Build is running'
+      dockerImages.each {
+        def hash = OsTools.runSafe(this, "docker build --quiet --no-cache --tag ${it.imageTag} -f ${it.dockerFilePath} .")
+        it.imageTag = hash.split(':')[1]
+
+        if (currentBuild.result == 'FAILURE') {
+            gitHub.statusUpdate commitId, 'failure', 'build', 'Build failed'
+            return
+        } else {
+            gitHub.statusUpdate commitId, 'success', 'build', 'Build succeeded'
+        }
+      }
+    }
+    if (params.nexus_repository_manager_version) {
+      stage('Commit Repository Manager Version Update') {
+        withCredentials([[$class: 'UsernamePasswordMultiBinding', credentialsId: 'integrations-github-api',
+                        usernameVariable: 'GITHUB_API_USERNAME', passwordVariable: 'GITHUB_API_PASSWORD']]) {
+          def commitMessage = params.nexus_repository_manager_version ? "Update Repository Manager to ${params.nexus_repository_manager_version}." : ""
```

Comment:
Ternary unnecessary since params.nexus_repository_manager_version always true

Reply:
In: 1626127